provider/resource_configuration.go

```diff
@@ -121,6 +121,13 @@ func resourceConfiguration() *schema.Resource {
 				},
 				Description: "Destination name and list of processor names to attach to the configuration. This option can be configured one or many times.",
 			},
+			"extensions": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    false,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "List of extensions names to attach to the configuration.",
+			},
```

Comment:
This provides the provider with a `[]any` which can be safely type asserted as a string. It is not guaranteed to be present, so the caller must check if it is nil.